modules/tinymce/src/core/test/ts/browser/EditorTest.ts

```diff
@@ -469,6 +469,18 @@ describe('browser.tinymce.core.EditorTest', () => {
     input.parentNode.removeChild(input);
   });
 
+  it('TINY-6946: Images should be properly cleaned up and not cause a crash', () => {
+    const editor = hook.editor();
+    editor.setContent('<img src="data:image/gif;base64,R0lGODdhIAAgAIABAP8AAP///ywAAAAAIAAgAAACHoSPqcvtD6OctNqLs968+w+G4kiW5omm6sq27gubBQA7AA==%A0">');
+    TinyAssertions.assertContent(editor, '<p><img src="data:image/gif;base64,R0lGODdhIAAgAIABAP8AAP///ywAAAAAIAAgAAACHoSPqcvtD6OctNqLs968+w+G4kiW5omm6sq27gubBQA7AA=="></p>');
+  });
+
+  it('TINY-6946: Images should cut off invalid data, even if the image remains invalid', () => {
```

Comment:
This doesn't look right? Generally our rule is to try to not lose data... so are you sure this is expected behaviour?

Reply:
To add to this, browsers don't cut off invalid data when parsing so this looks like a bug to me.

Reply:
Ahh, it's probably because we convert it to a blob and then convert back to base64 content. In which case it'd be expected since the blob can't contain invalid data. If that is the reason, then can you please add a comment about why it's expected we're losing content?

Reply:
This is because the base64 parser will parse valid things as long as it can then convert that to a blob url and replace the whole attribute contents with that new blob url. So anything that isn't valid base64 after the valid one will be thrown away. In theory we could try to retain the invalid data here but not sure what the point of that would be exactly since the browser will not render it anyway since it's corrupt.
So it's kind of cleaning up invalid base64 urls by accident but that is what the customer requested here so we just add a test so that we don't break this in the future.